pkg/operator/ceph/object/spec.go

```diff
@@ -458,6 +464,12 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) (v1.Container,
 		container.Args = append(container.Args, cephconfig.NewFlag(flag, val))
 	}
 
+	if c.isLocalizedReadAffinitySet() {
+		logger.Debug("localized readAffnity is set. Update rgw container command")
+		container.Args = append(container.Args, cephconfig.NewFlag("crush location", fmt.Sprintf("%s=%s", "host", "$hostName")))
+		container.Command = []string{"bash", "-x", "-c", `hostName=${NODE_NAME//./-} && exec radosgw "$@"`}
```

Comment:
I think this is equivalent and doesn't require splitting logic between command and args. I think this will be easier to interpret when debugging and reading the deployed pod spec. 

```suggestion
		container.Command = []string{"bash", "-x", "-c", `exec radosgw --host="${NODE_NAME//./-}" "$@"`}
```

Reply:
two arguments against this: 
- We already set the `host` to pod name [here](https://github.com/rook/rook/blob/6189a7be9062098aaf19a91adc128b361e1cd507/pkg/operator/ceph/object/spec.go#L380).  And I think we discussed not to change that. 
- When/if we decided to add support for portable osds, we would need to use the `--crush-location`.

Reply:
how about `container.Command = []string{"bash", "-x", "-c", `exec radosgw --crush-location="host=${NODE_NAME//./-}" "$@"`}` ?

Reply:
Oh, I see. I missed that. I think maybe we need to remove the old value of `host` when this feature is enabled then. Having it get set twice will take up the last value in the list IIRC, but it might be a bit confusing for anyone inspecting the pod.

Reply:
Now I'm only adding `host` when localized read is not set.